cluster/src/main/scala/org/apache/spark/util/SnappyUtils.scala

```diff
@@ -50,6 +50,19 @@ object SnappyUtils {
     }
   }
 
+  def removeJobJar(sc: SparkContext, jarName : String): Unit = {
+    def getName(path: String): String = new File(path).getName
+    val keyToRemove = sc.listJars().filter(getName(_) == getName(jarName))
+    if (keyToRemove.nonEmpty) {
+      val callbacks = ToolsCallbackInit.toolsCallback
+      //@TODO This is a temp workaround to fix SNAP-1133. sc.addedJar should be directly be accessible from here.
+      //May be due to scala version mismatch.
+      if(callbacks != null){
```

Comment:
Yes. That's the reason why I have added jar remove code in streaming context stop() method of SnappyStreamingContext , which we are extending in case of jobs.